simulators/portal-interop/src/main.rs

```diff
@@ -237,574 +209,104 @@ dyn_async! {
-        let header_value: HistoryContentValue =
-            serde_yaml::from_value(values[4].get("content_value").unwrap().clone()).unwrap();
-        let content_key: HistoryContentKey =
-            serde_yaml::from_value(values[6].get("content_key").unwrap().clone()).unwrap();
-        let content_value: HistoryContentValue =
-            serde_yaml::from_value(values[6].get("content_value").unwrap().clone()).unwrap();
-        test_offer_x(client_a, client_b, (content_key, content_value), Some((header_key, header_value))).await;
-    }
-}
+        let target_enr = match client_b.rpc.node_info().await {
+            Ok(node_info) => node_info.enr,
+            Err(err) => {
+                panic!("Error getting node info: {err:?}");
+            }
+        };
 
-dyn_async! {
-    async fn test_offer_receipts_block_15600000<'a> (client_a: Client, client_b: Client) {
-        let values = std::fs::read_to_string("./test-data/test_data_collection_of_forks_blocks.yaml")
-            .expect("cannot find test asset");
-        let values: Value = serde_yaml::from_str(&values).unwrap();
-        let header_key: HistoryContentKey =
-            serde_yaml::from_value(values[7].get("content_key").unwrap().clone()).unwrap();
-        let header_value: HistoryContentValue =
-            serde_yaml::from_value(values[7].get("content_value").unwrap().clone()).unwrap();
-        let content_key: HistoryContentKey =
-            serde_yaml::from_value(values[9].get("content_key").unwrap().clone()).unwrap();
-        let content_value: HistoryContentValue =
-            serde_yaml::from_value(values[9].get("content_value").unwrap().clone()).unwrap();
-        test_offer_x(client_a, client_b, (content_key, content_value), Some((header_key, header_value))).await;
-    }
-}
+        let result = client_a.rpc.find_content(target_enr, header_with_proof_key.clone()).await;
 
-dyn_async! {
-    async fn test_offer_receipts_block_17510000<'a> (client_a: Client, client_b: Client) {
-        let values = std::fs::read_to_string("./test-data/test_data_collection_of_forks_blocks.yaml")
-            .expect("cannot find test asset");
-        let values: Value = serde_yaml::from_str(&values).unwrap();
-        let header_key: HistoryContentKey =
-            serde_yaml::from_value(values[10].get("content_key").unwrap().clone()).unwrap();
-        let header_value: HistoryContentValue =
-            serde_yaml::from_value(values[10].get("content_value").unwrap().clone()).unwrap();
-        let content_key: HistoryContentKey =
-            serde_yaml::from_value(values[12].get("content_key").unwrap().clone()).unwrap();
-        let content_value: HistoryContentValue =
-            serde_yaml::from_value(values[12].get("content_value").unwrap().clone()).unwrap();
-        test_offer_x(client_a, client_b, (content_key, content_value), Some((header_key, header_value))).await;
+        match result {
```

Comment:
```rust
if let Ok(ContentInfo::Enrs { enrs: val }) = result {
    if !val.is_empty() {
        panic!("Error: Unexpected FINDCONTENT response: expected ContentInfo....");
    }
} else {
    panic!("Error....");
}
```

warning: I didn't test the above, but I think this might help get rid of some of these lines, but only if you don't think it's important to have all of these different error messages for debugging

Reply:
I think it would be helpful if someone was debugging it so I will leave it now it is